src/ensembl/src/content/app/browser/services/browser-messaging-service/browser-messaging-service.ts

```diff
@@ -70,11 +89,16 @@ export class BrowserMessagingService {
     }
     const subscribers = this.subscribers[type];
     if (subscribers) {
-      subscribers.forEach((subscriber: Function) => subscriber(payload));
+      subscribers.forEach((subscriber: (payload: any) => void) =>
+        subscriber(payload)
+      );
     }
   };
 
-  public subscribe = (eventName: string, callback: Function) => {
+  public subscribe = (
+    eventName: string,
+    callback: (...args: any[]) => void
```

Comment:
Sorry for the late realisation, @msmanoj, but, compared to what we have in dev (https://github.com/Ensembl/ensembl-client/blob/500e9a4f1969020d9e8290617850202421747c05/src/ensembl/src/content/app/browser/zmenu/ZmenuController.tsx#L49-L53), we are of course losing the power of placing subscribers only in those arrays that concern them, and, as you rightly pointed out, we'll be notifying all subscribers now.

I think we can do better. I think we can possibly use the `action` part of the payload as a key for `this.subscribers`. The question then becomes how to handle cases when a single function wants to subscribe to multiple actions — what the signature of the `subscribe` method should be. One possibility is to make it capable of handling both a string and an array of strings as the first argument. An additional possibility is to make it capable of handling wildcards, as done [here](https://sahadar.github.io/pubsub/index.html#subscribe-wildcard-a-name-subscribe_wildcard-a-); but that may be too much. Will sure be hell of a fun to type-check!

Reply:
The required changes are made in the following commit:
https://github.com/Ensembl/ensembl-client/pull/334/commits/126cdc88fd33a53b48d985ba59dbad364dd36f3b